buildSrc/src/test/kotlin/com/datadog/gradle/plugin/jsonschema/ModelValidationTest.kt

```diff
@@ -96,21 +102,83 @@ class ModelValidationTest(
         }
     }
 
+    private val mapTypeComparator = Comparator<Map<*, *>> { t1, t2 ->
+        if (t2.size == t1.size) {
+            val mismatches = t1.filter { (k, v1) ->
+                val v2 = t2[k]
+                compareMapValues(v1, v2)
+            }
+            mismatches.size
+        } else {
+            1
+        }
+    }
+
+    private val informationComparator = Comparator<Company.Information> { t1, t2 ->
+        if (t1.date != t2.date) {
+            -1
+        } else if (t1.priority != t2.priority) {
+            -2
+        } else {
+            mapTypeComparator.compare(t1.additionalProperties, t2.additionalProperties)
+        }
+    }
+
+    private fun compareMapValues(v1: Any?, v2: Any?): Boolean {
+        return if (v1 is JsonElement) {
+            compareJsonElement(v1, v2)
+        } else if (v1 is Map<*, *> && v2 is Map<*, *>) {
+            mapTypeComparator.compare(v1, v2) == 0
+        } else {
+            v2 != v1
+        }
+    }
+
+    private fun compareJsonElement(
+        v1: JsonElement,
+        v2: Any?
+    ): Boolean {
+        return when (v2) {
+            null -> v1 != JsonNull.INSTANCE
+            is Boolean -> v1.asBoolean != v2
+            is Int -> v1.asInt != v2
+            is Long -> v1.asLong != v2
+            is Float -> v1.asFloat != v2
+            is Double -> v1.asDouble != v2
+            is String -> v1.asString != v2
+            is Date -> v1.asLong != v2
```

Comment:
here we are comparing `Long` and `Date`, is this a valid comparison? probably `Long` is backed by the timestamp then?

maybe it should be `v2.time` here?